### ISSUE TITLE
stuttering/distorted sound on iOS 14.4 iPhone 8plus -> iPhone 12

### ISSUE THREAD

**Issue by bwillner** [twilio/twilio-video.js#1366]
- [ x] I have verified that the issue occurs with the latest twilio-video.js release and is not marked as a known issue in the [CHANGELOG.md](https://github.com/twilio/twilio-video.js/blob/master/CHANGELOG.md).
 - [x ] I reviewed the [Common Issues](https://github.com/twilio/twilio-video.js/blob/master/COMMON_ISSUES.md) and open GitHub issues and verified that this report represents a potentially new issue.
 - [ ] I verified that the [Quickstart](https://github.com/twilio/video-quickstart-js) application works in my environment.
 - [ x] I am not sharing any [Personally Identifiable Information (PII)](https://www.twilio.com/docs/glossary/what-is-personally-identifiable-information-pii)
       or sensitive account information (API keys, credentials, etc.) when reporting this issue.

**Code to reproduce the issue:**

```js
// TODO
```

**Expected behavior:**

consistent audio quality.

**Actual behavior:**

Hi,

the connection seems to be ok for the first few minutes. After that both parties here crackling sounds and distortions. this gets worse over time. If the call is ended and a new call is started the issue seems to start immediately.

Here is a room ID: RM5c334ded20d15e233c42af6d00ad17ef
tested on iPhone 8plus (14.4) -> iPhone 12 (14.4)

Here are other room IDs:
RM8f28805ad5f54477f904afa0162dc10f
RM07d87d5dd89dca2ecc3128c56068e190
tested on iPhone 8plus (14.4) -> iPad Air (14.1)

Please let me know if you need any other logs etc. 

Thanks for your help in advance!

TODO

**Software versions:**

 - [x ] Browser(s): iOS Safari
 - [ x] Operating System: iOS 14.4
 - [ x] twilio-video.js: 2.10.0
 - [ x] Third-party libraries (e.g., Angular, React, etc.): React

**Comment by bwillner** [comment]
Is there any status on this? this bug is making our app unusable for iPhone users at the moment.

Hoping to get this sorted out.

Thanks

**Comment by makarandp0** [comment]
Hello @bwillner, Thank you for reporting this issue. I have not been able to repro it so far (but I do not have exact phone versions you mentioned). You mentioned two scenarios where this issue reproduces. Have you been able to narrow it down to specific phone/version? Is it caller version or callee version that is an issue? If you replace caller (or callee) with desktop client does the issue still repro?

Any details to narrow repro would that you could provide would help al lot.

Thanks,
Makarand

**Comment by bwillner** [comment]
Hi @makarandp0, in our tests we are experiencing this issue only with iPhone 8plus. We have tested two different iPhones 8pluses and both have the issue. We have tested with other iPhones and iPads, but unfortunately we don't have that many different iOS devices to test with.

- The issue appears on the iPhone side. The other side can sometimes here the crackling as well. 
- It happens every call if at least one device is an iPhone.
- iPhone to Desktop (both Chrome & Safari) has the same problem.

Please let me know if I can provide any other information.

Thanks for your efforts

**Comment by bwillner** [comment]
Hi, do you have any progress about this?

We have done some further testing. The problem might be related to overall cpu load. If the video resolution is reduced or no video is used, the audio is more stable.

Weirdly the audio problems are triggered e.g. by switching the camera or turning the camera or mic on/off. Could this be related to #1367 ?

Unfortunately I can't find much information about how to properly discard unused video/audio tracks. At the moment we are calling .detach() and .stop() on the localVideoTrack and localAudioTrack as well as .unpublishTrack() on localParticipant. Is there anything to do to avoid resources to be used in the background?

Thanks

**Comment by YakirSolomon** [comment]
Hi @bwillner, did you happen to solve the issue? facing the same issue on my app as well.
Thanks.

**Comment by gabrieljbaker** [comment]
I notice this issue on an iphone 12 plus, safari.

**Comment by makarandp0** [comment]
Hello @bwillner @gabrieljbaker  - Sorry for long delay - We recently found about a [bug in webrtc dtx implementation](https://bugs.chromium.org/p/webrtc/issues/detail?id=13051) that can lead to background noise on a muted audio track.  twilio-video.js enables DTX (discontinuous transmission) by default for opus. I am wondering if this bug is leading to your issue. To workaround this issue, you can disable the dtx with:

const { connect } = require('twilio-video');

// Disable DTX for Opus.
connect('token', {
  preferredAudioCodecs: [{ codec: 'opus', dtx: false }]
});

Please let us know if you still seeing the audio issue and if the above workaround helps.
Thanks,
Makarand

**Comment by cindyloo** [comment]
potential workaround from other related issues:
https://github.com/twilio/twilio-video.js/issues/1366

**Comment by sarahtully** [comment]
We experienced this issue as well. After 5 minutes the iOS participant's audio has static. When they unplug headphones after another ~5 minutes the static occurs again.

Room sid: RM018334cb0cb29d2b9bd534d7f51437b0

User was on an iPadOS (10.15.6) Safari 15.4
